core.d.ts

```diff
@@ -435,3 +438,93 @@ console.log(await fileTypeFromBlob(blob));
+
+Example detector array which can be extended and provided via the fileTypeOptions argument:
+
+```js
+const customDetectors = [
+	async tokenizer => {
+		const unicornHeader = [85, 78, 73, 67, 79, 82, 78]; // "UNICORN" as decimal string
+		const buffer = Buffer.alloc(7);
+		await tokenizer.peekBuffer(buffer, {length: unicornHeader.length, mayBeLess: true});
+		if (unicornHeader.every((value, index) => value === buffer[index])) {
+			return {ext: 'unicorn', mime: 'application/unicorn'};
+		}
+		return undefined;
+	}
+]
+
+	Example usage:
+
+```js
+const buffer = ...
+const parser = new FileTypeParser({customDetectors});
+const fileType = await parser.fromBuffer(buffer);
+
+	fromStream(...), fromTokenizer(...), fromBlob(...) and stream(...) are available in the same manner.
+
+ @param tokenizer - An [`ITokenizer`](https://github.com/Borewit/strtok3#tokenizer) usable as source of the examined file.
+ @param fileType - FileTypeResult detected by the standard detections or a previous custom detection. Undefined if no matching fileTypeResult could be found.
+ @returns supposedly detected file extension and MIME type as a FileTypeResult-like object, or `undefined` when there is no match.
+*/
+export type Detector = (tokenizer: ITokenizer, fileType?: FileTypeResult) => Promise<FileTypeResult | undefined>;
+
+export type FileTypeOptions = {
+	customDetectors?: Iterable<Detector>;
+};
+
+export declare class TokenizerPositionError extends Error {
+	constructor(message?: string);
+}
+
+export declare class FileTypeParser {
+	detectors: Iterable<Detector>;
+
+	constructor(options?: {customDetectors?: Iterable<Detector>});
+
+	/**
+	 *
+	 * Works the same way as {@link fileTypeFromBuffer}, additionally taking into account custom detectors (if any were provided to the constructor).
+	 */
+	fromBuffer(buffer: Uint8Array | ArrayBuffer): Promise<FileTypeResult | undefined>;
+
+	/**
+	 *
+	 * Works the same way as {@link fileTypeFromStream}, additionally taking into account custom detectors (if any were provided to the constructor).
+	 */
+	fromStream(stream: ReadableStream): Promise<FileTypeResult | undefined>;
+
+	/**
+	 *
+	 * Works the same way as {@link fileTypeFromTokenizer}, additionally taking into account custom detectors (if any were provided to the constructor).
+	 */
+	fromTokenizer(tokenizer: ITokenizer): Promise<FileTypeResult | undefined>;
+
+	/**
+	 *
+	 * Works the same way as {@link fileTypeFromBlob}, additionally taking into account custom detectors (if any were provided to the constructor).
+	 */
+	fromBlob(blob: Blob): Promise<FileTypeResult | undefined>;
+
+	/**
+	 *
+	 * Works the same way as {@link fileTypeStream}, additionally taking into account custom detectors (if any were provided to the constructor).
+	 */
+	stream(readableStream: ReadableStream, options?: StreamOptions): Promise<FileTypeResult | undefined>;
```

Comment:
Suggestion, maybe good to articulate the different functionality here.
```suggestion
	toDetectingStream(readableStream: ReadableStream, options?: StreamOptions): Promise<FileTypeResult | undefined>;
```
Same comment provided for the implementation

Reply:
Done.